File: dbsgur/leetcode/string/49.py
strs = ["eat", "tea", "tan", "ate", "nat", "bat"]
dict = {}
len_str = len(strs)
for i in range(len_str):
    tmp = ''.join(sorted(list(strs[i])))
    # print(list(strs[i]))
    if tmp in dict:
        dict[tmp].append(strs[i])
    else:
        dict[tmp] = [strs[i]]

    # 키 : [원래문자] value : [문자들]
    # 같으면 반환
# print(dict.values())
answer = list(dict.values())
print(answer)

File: dbsgur/leetcode/string/49.js
var groupAnagrams = function (strs) {
  let obj = new Map();
  let tmp;
  for (str of strs) {
    tmp = Array.from(str).sort().join("");
    if (obj.has(tmp)) {
      obj.get(tmp).push(str);
    } else {
      obj.set(tmp, [str]);
    }
  }
  return [...obj.values()];
};
